test/extensions/stats_sinks/metrics_service/grpc_metrics_service_impl_test.cc

```diff
@@ -298,24 +305,86 @@ TEST_F(MetricsServiceSinkTest, FlushPredicate) {
 
   // Default predicate only accepts used metrics.
   {
-    MetricsFlusher flusher(true, true);
+    MetricsFlusher flusher(true, true,
+                           envoy::config::metrics::v3::HistogramEmitMode::SUMMARY_AND_HISTOGRAM);
     auto metrics = flusher.flush(snapshot_);
     EXPECT_EQ(1, metrics->size());
   }
 
   // Using a predicate that accepts all metrics, we'd flush both metrics.
   {
-    MetricsFlusher flusher(true, true, [](const auto&) { return true; });
+    MetricsFlusher flusher(true, true,
+                           envoy::config::metrics::v3::HistogramEmitMode::SUMMARY_AND_HISTOGRAM,
+                           [](const auto&) { return true; });
     auto metrics = flusher.flush(snapshot_);
     EXPECT_EQ(2, metrics->size());
   }
 
   // Using a predicate that rejects all metrics, we'd flush no metrics.
-  MetricsFlusher flusher(true, true, [](const auto&) { return false; });
+  MetricsFlusher flusher(true, true,
+                         envoy::config::metrics::v3::HistogramEmitMode::SUMMARY_AND_HISTOGRAM,
+                         [](const auto&) { return false; });
   auto metrics = flusher.flush(snapshot_);
   EXPECT_EQ(0, metrics->size());
 }
 
+// Test that verifies counters are reported as the delta between flushes when configured to do so.
+TEST_F(MetricsServiceSinkTest, HistogramEmitMode) {
+  addHistogramToSnapshot("test_histogram");
+
+  {
```

Comment:
nit: just break these up 3 three separate test methods. You already have the common prep factored to one nice helper function.